compliance_suite/utils/test_utils.py

```diff
@@ -23,7 +23,9 @@ def tag_matcher(
         present in the yaml_tags. Otherwise, False.
     """
 
+    # If include_tags is empty, all tests are considered included
+
     return (
         not any(exclude_tag in yaml_tags for exclude_tag in exclude_tags)
-        and any(include_tag in yaml_tags for include_tag in include_tags)
+        and (not include_tags or any(include_tag in yaml_tags for include_tag in include_tags))
```

Comment:
Contrary to what we discussed, I used this approach where we assume all tests are included by default. The elimination will be decided depending on the `--exclude-tags`.
This also allows us to handle tests which have no tags (otherwise the check was failing for empty `yaml_tags`)

Reply:
We initially considered allowing a test to run if neither `--include-tags` nor `--exclude-tags` were provided. In that case, the code change would have been:
```
if include_tags == [] and exclude_tags ==[]:
  return True
```
This would have worked perfectly, but we also updated the tags of each test. Some tests, specifically functional tests, no longer have any tags assigned to them, resulting in an empty `yaml_tags` list. Therefore, both conditions in `return` statement evaluated to `False`, leading to the simplified expression `return (not False and False)`, i.e. `return False`. Consequently, the functional tests were never executed. Handling this situation would have required unnecessary additional changes.

Therefore, I modified the code to assume a test is included by default if `--include-tags` is empty, which actually makes sense. We can then proceed to check the exclusion condition.

In summary, rather than touching `--exclude-tags`, I updated the condition based solely on `--include-tags`. The exclusion mechanism remains the same: to exclude a test, `--exclude-tags` must be provided and matched. For inclusion, if `--include-tags` is provided, it needs to match the test tags. Otherwise, the test is included by default.

Reply:
Earlier, the execution of all tests was handled by the tag `all`. After its removal, we needed a way to execute all tests and thus the above way was implemented.

Reply:
I'm sorry, but I still don't get it 🙈😅 If I want to run all test, do I need to specify something or not?

Reply:
No problem :smile:.  No, you dont need to specify anything to run all tests.

> Condition to run all the tests:
> If --include-tags is not provided, all tests will be executed by default (subject to exclusion criteria if --exclude-tags is provided). Example: report --server SERVER --version VERSION will run all available tests